### DISCUSSION TITLE
Thank you!

### DISCUSSION THREAD

**Discussion by DarrenPIngram** [andrewbanchich/shreddit#75] [General]
With the recent nonsense on Reddit, I felt it was valuable to extend a thank you for you due to your work with this application, especially with its ability to add the "GDPR" file in to the mix. 

Oh, and in keeping with the trend de jour on Reddit [well it was before I left]... F /u/Spez :)

**Comment by andrewbanchich** [comment]
Thank you for saying so! I'm really happy people are making good use of it.

**Reply by photonwave** [reply]
Hi Andrew, for all the users out there, including myself, who aren't familiar or so familiar with coding, could you make a video or some other kind of guide for getting your amazing app to run on Reddit? Thank you :)

**Comment by DarrenPIngram** [comment]
> Hi Andrew, for all the users out there, including myself, who aren't familiar or so familiar with coding, could you make a video or some other kind of guide for getting your amazing app to run on Reddit? 

I'll just butt in and say it might be a big ask - so many potential variables and there is already a way to install it for the unfamiliar with cargo [rust package manager] if you are using Linux. It is unclear if you are even using Linux [forgive me if I've made a wrong assumption - your wording just had me wondering if you had thought it to be a Windows app or something]. 

If you do get it installed then it is a powerful application which it does what it says on the tin!

**Reply by photonwave** [reply]
I know that I and likely many others would appreciate a tutorial for how to install it. I also know that without one, I won’t be able to install it because I’m not a programmer. I have access to a PC so I could install it if I had detailed instructions. Again, app sounds amazing @andrewbanchich :D

**Reply by DarrenPIngram** [reply]
Well there IS already a way to install it under cargo, assuming you have that installed on your system. 

You've looked at the binaries? https://github.com/andrewbanchich/shreddit/releases

What exact OS are you running? Even if you get it installed you then need to craft the API and command strings.

I am not speaking on behalf of the developer, and he might do as you ask, but to be honest it feels a big ask and your message is still massively vague too.  I don't want to do you a disservice, but it feels you may struggle even if you get it installed, based on this exchange.

I'll leave it here as it is the developer's space - just trying to shape your expectations and give you a tip or two, but that hasn't got through.

**Reply by photonwave** [reply]
I understood your message very clearly and your point about providing more specificity. I am running Windows 11.

About my request being a big ask. I am sure @andrewbanchich is aware that he isn’t obligated to make any tutorials just because a user asked for one. Shreddit sounds incredibly useful and powerful for a website that’s insanely popular. I’m sure that of the many thousands of people who would probably find this app useful, there is a good percentage who aren’t programmers like myself. My request is for all of those users, of which I’m sure there are many and for myself.

**Reply by DarrenPIngram** [reply]
Well this is [to me] a problem - It took a lot of gos to even get what OS you were using and still it seems you are not reading or understanding the questions asked of you. I've even pointed you to the binaries page and you will find a binary for Windows there - but even that has gone past you... The material is there -- except a step-by-step hand-holding guide [and even then "you'd" need more than just how to install the application. 

Anyway, I'll leave it up to the developer, I've tried to point you but in vain it seems. Respectfully, please read the information given and read several times. It might be worth you even asking on "Fiverr" or similar platforms for some assistance if you need it, and if the developer is understandably unable or unwilling to provide it. I doubt he has made a nice GUI version expressly for Windows [I don't use Windows to verify this] and it still then requires some effort by the user. It is not [at least on Linux] a "fill in your user name, enter API key and press return" package I could give to my father [that is not a criticism, just a fact].

I desperately don't want this reply to be harsh or feel condescending as that is not the motive. I just fear you've discovered this application and are not appreciating what it actually is [other than its objective]. There are GUI-type programs out there [even a web hosted service] that claim to do similar tasks to this application - but I have not verified their status today and don't know if they accept the Reddit data file either. It has been a bit of a cat-and-mouse breaking game by Reddit, perhaps for obvious reasons, at least from what I recall from reading about a browser extension in the past. 

Now this is my final word, as I definitely do not speak on behalf of the developer, but I just tried in good faith to give some guidance.

**Reply by andrewbanchich** [reply]
Hi @calmtake , I appreciate the kind words 😊

@DarrenPIngram is pretty much on the mark as to why I'd be apprehensive about creating a video, although I do also completely understand your point. Videos can help a great deal with clarifying how things work, especially if you're unfamiliar with some terminology in the docs.

Unfortunately, the issue is that the ways to install and run binaries are different per OS, so I'd need to make a video for each one (and I don't use Windows).

This is a CLI app, so it should be fairly easy to use. However, not everyone is familiar with CLI apps, so if you aren't, I'd recommend learning about them a little and (hopefully) the README should become clear. I also would recommend downloading the exe directly from the Releases page @DarrenPIngram linked to, since that's the easiest way. No need to worry about what cargo is or anything.

If there is something in the README that is unclear, just let me know and I can see if I can improve it!

**Reply by photonwave** [reply]
Thank you @andrewbanchich for your detailed response and help. So, I do really want to be able to use your app but will doing that on a computer running Windows 11 mean I have to do any coding? And if so, could you possibly post the code necessary for a Windows computer running Windows 11 to install it where it needs to be installed?

Thank you.

**Reply by andrewbanchich** [reply]
No coding necessary at all!

**Reply by DarrenPIngram** [reply]
At the risk of confusing @calmtake, since I don't use the Windows version, I presume the Windows version is equally a CLI tool. If so, seeing how I run it on Linux may allow him/her to customise an argument. It can vary [I had it in a script, but the basic command line is]: 

[a link to the Shreddit code/app]   --username "$USERNAME"  --password "$PASSWORD"  --client-id "$CLIENT_ID" --client-secret "$CLIENT_SECRET"  --gdpr-export-dir "/link_to_the_file_directory"

Obviously replace $USERNAME etc [as these were variables] with your own information. You will need to create the client ID etc [can't do that for you...] and if you don't have a GDPR file ignore the last bit.

Remember if you want to test what you are doing without damage first add --dry-run to the end. 

There can well be other variables [read the documentation, I don't remember]. The above is what worked for me as I wanted to nuke everything.

**Reply by photonwave** [reply]
Hi @andrewbanchich, not sure if my reply went through so I'm replying here directly. Thank you for your update. I'm able to download your app on a mac.

So it sounds like I need to use Terminal for this app to run in Reddit. I have used Terminal before but I'm still a novice at it. I'm looking for clarification on your instructions from you README.md file. I understand that you're saying put in the following code into the CLI:

"Usage: shreddit [OPTIONS] --username <USERNAME> --password <PASSWORD> --client-id <CLIENT_ID> --client-secret <CLIENT_SECRET>

Options:
  -u, --username <USERNAME>..."

My question is, after downloading shreddit-mac to my downloads folder, what should I be doing in Terminal to get to that step I quoted above?

Do I need to direct the CLI to a specific file that I downloaded or do I need to navigate it to somewhere else?

Sorry that this is such a beginner's question, but I would really appreciate some clarification on this part. Thank you!

**Reply by DarrenPIngram** [reply]
I can answer in the interim [unless a new version packaged for the Mac has appeared that I've not noticed].

You would need to use terminal to fire the commands, make the necessary configuration/pass on the Reddit credentials.  

I think I posted a while back a script that I used [on Linux but the concept should work under MacOS]. 

Maybe look back before the developer has chance to reply, have a try with that.

**Comment by aceryz** [comment]
Hi Andrew!

Another big "thank you" from me. It was somewhat cathartic to see all the comments and posts being deleted one by one, haha.

Thanks!

**Reply by andrewbanchich** [reply]
Thank YOU! 🙏
